public/app/types/series.ts

```diff
@@ -87,5 +88,14 @@ export interface DataQueryOptions {
 }
 
 export interface DataSourceApi {
+  /**
+   * Imports queries from a different datasource
+   */
+  importQueries?(queries: string[], origin: DataSource): Promise<string[]> | string[];
```

Comment:
queries cannot be just a string, needs to be an object with properties, for example: 

```typescript
interface Query {
  refId: string;  
  [key: string]: any 
}
```

Reply:
Alternate name suggestion for the function would be `translateQueries`  but not sure if it's better. Just import feels like a persisted action and not a transformation function.

Reply:
If it's alright I'd address query objects in a different PR. Also, I couldnt find any other use of the `Query` type, is that new?

Reply:
Sorry, the interface was called DataQuery:

https://github.com/grafana/grafana/blob/master/public/app/types/series.ts#L71

Reply:
So before merging this PR would not be good to change this to DataQuery? (And the return type). So we don't add stuff this interface that is specific for prometheus. You dont need to refactor the whole of Explore to use object queries for this part I think. 

Also origin is still the wrong type here. It should also be DataSourceApi type. If you just add meta: PluginMeta to DataSourceApi you can use that everywhere in Explore instead of any I think.